ext/curl/tests/gh15547.phpt

```diff
@@ -0,0 +1,19 @@
+--TEST--
+GH-15547 - overflow on timeout argument
+--EXTENSIONS--
+curl
+--FILE--
+<?php
+
+$mh = curl_multi_init();
+var_dump(curl_multi_select($mh, -2500000));
```

Comment:
If we can assume `INT_MAX == 0x7ffffff` for all supported architectures (and I think we can), this looks good.